src/LiveComponent/doc/index.rst

```diff
@@ -2523,41 +2523,136 @@ To change the initial tag from a ``div`` to something else, use the ``loading-ta
 
-      <div
-          {{ attributes }}
-    +     data-poll
-      >
+Basic usage
+^^^^^^^^^^^
 
-This will make a request every 2 seconds to re-render the component. You
-can change this by adding a ``delay()`` modifier. When you do this, you
-need to be specific that you want to call the ``$render`` method. To
-delay for 500ms:
+By default, polling runs the ``$render`` action every 2000 milliseconds (2 seconds).
 
 .. code-block:: html+twig
 
-    <div
-        {{ attributes }}
-        data-poll="delay(500)|$render"
-    >
+    {# Unlimited polling every 2 seconds (default) using $render #}
+    <div {{ attributes }} data-poll>...</div>
+
+    {# Poll every 5 seconds, up to 20 times, using $render #}
+    <div {{ attributes }} data-poll="delay(5000)|limit(20)|$render">...</div>
+
+    {# Poll a custom action (savePost) every 3 seconds, up to 10 times #}
+    <div {{ attributes }} data-poll="delay(3000)|limit(10)|savePost">...</div>
+
+Available Modifiers
+^^^^^^^^^^^^^^^^^^^
+
+- ``delay(ms)`` — The delay between polls in milliseconds (default: ``2000``)
+- ``limit(n)`` — Maximum number of times to run the poll (default: unlimited)
+- ``actionName`` — The component action to call (default: ``$render``)
+
+Poll Hooks
+^^^^^^^^^^
 
-You can also trigger a specific "action" instead of a normal re-render:
+The component emits lifecycle hooks during polling. You can listen to these using JavaScript, for example:
+
+.. code-block:: javascript
+
+    // controllers/poll_controller.js
+    import { Controller } from '@hotwired/stimulus';
+    import { getComponent } from '@symfony/ux-live-component';
+
+    export default class extends Controller {
+        async connect() {
+            this.component = await getComponent(this.element);
+
+            // Disable default error window (optional)
+            this.component.on('response:error', (backendResponse, controls) => {
+                controls.displayError = false;
+            });
+
+            this.component.on('poll:started', ({ actionName, limit }) => {
+                console.log(`Polling started: ${actionName}, limit: ${limit}`);
+            });
+
+            this.component.on('poll:running', ({ actionName, count, limit }) => {
+                console.log(`Polling running: ${actionName} (${count}/${limit})`);
+            });
+
+            this.component.on('poll:paused', ({ actionName, count, limit }) => {
+                console.log(`Polling paused: ${actionName}`);
+            });
+
+            this.component.on('poll:stopped', ({ actionName, finalCount, limit }) => {
+                console.log(`Polling stopped: ${actionName}, total runs: ${finalCount}`);
+            });
+
+            this.component.on('poll:error', ({ actionName, finalCount, limit, errorMessage }) => {
+                console.error(`Polling error on ${actionName}: ${errorMessage}`);
+            });
+        }
+    }
+
```

Comment:
Do we need the whole block here ? Maybe one block and then comments would be enough ? 

Something like that ? 

```js
  this.component.on('poll:started', ({ actionName, limit }) => {
      console.log(`Polling started: ${actionName}, limit: ${limit}`);
  });
  
  // this.component.on('poll:started', ({ actionName, limit }) => { ... 
  // this.component.on('poll:running', ({ actionName, count, limit }) => { ... 
  // this.component.on('poll:paused', ({ actionName, count, limit }) => { ... 
  // this.component.on('poll:stopped', ({ actionName, finalCount, limit }) => { ...
  // this.component.on('poll:error', ({ actionName, finalCount, limit, errorMessage }) => { ... 
```